Public/Src/Cache/ContentStore/Distributed/NuCache/EventStreaming/ContentLocationEventData.cs

```diff
@@ -90,6 +93,8 @@ public static ContentLocationEventData Deserialize(BuildXLReader reader, DateTim
                     return new TouchContentLocationEventData(sender, hashes, eventTimeUtc);
                 case EventKind.Reconcile:
                     return new ReconcileContentLocationEventData(sender, reader.ReadString());
+                case EventKind.AddLocationWithoutTouching:
+                    return new AddContentLocationWithoutTouchingEventData(sender, hashes, reader.ReadReadOnlyList(r => r.ReadInt64Compact()));

```

Comment:
To avoid any runtime issues we should make sure that we won't send this events until all the masters are updated.

The bits rollout is happening randomly, so if the feature that causes this events to be sent are on by default it is possible that some workers will be updated first, generate this event and will send it to the old master.

Reply:
It will just be a matter of keeping this disabled until rollout is complete. It's disabled by default, so I don't think there is much risk, as long as we leave a reasonable period between release and enabling the feature

Reply:
It will just be a matter of keeping this disabled until rollout is complete. It's disabled by default, so I don't think there is much risk, as long as we leave a reasonable period between release and enabling the feature

---
In reply to: [342735439](https://github.com/microsoft/BuildXL/pull/1141#discussion_r342735439) [](ancestors = 342735439)